tests/conftest.py

```diff
@@ -331,6 +331,30 @@ def spark_session(test_backends):
         raise ValueError("spark tests are requested, but pyspark is not installed")
 
 
+@pytest.mark.order(index=2)
+@pytest.fixture(scope="module")
+def spark_warehouse_session(test_backends, tmp_path_factory):
+    if "SparkDFDataset" not in test_backends:
+        pytest.skip("No spark backend selected.")
+
+    try:
+        import pyspark
+        from pyspark.sql import SparkSession
+    except ImportError:
+        raise ValueError("spark tests are requested, but pyspark is not installed")
+
+    spark_warehouse_path = str(tmp_path_factory.mktemp("spark-warehouse"))
+    spark = get_or_create_spark_application(
```

Comment:
@lccasagrande Please use type hints for all variables throughout the code.  Thank you.

Reply:
Fixed